tooling/template-base/components/LearnerAccess/LoadUserLearning.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState } from 'react';
+import listViewSelector from '../../renderer/listViewSelector.svg';
+import gridSelected from '../../renderer/gridSelected.svg';
+
+import { LoadedComponentProps } from './Types/types';
+import {
+  useUserContentItemsQuery,
+  LoadingDots,
+  hydrateContent,
+  HydratedContentItem
+} from '@thoughtindustries/content';
+import { useTranslation } from 'react-i18next';
+import LearnerAccessGridView from './Views/GridView';
+import LearnerAccessDisplayListView from './Views/DisplayListView';
+
+const LoadUserLearning = ({ query, kind, sort }: LoadedComponentProps): JSX.Element => {
+  const [gridViewActive, setGridActive] = useState(true);
+
+  // update state to display grid only on mobile
+  const handleResize = () => {
+    if (window.innerWidth < 600) {
+      setGridActive(true);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+  });
+
+  const { data, loading, error } = useUserContentItemsQuery({
+    variables: {
+      query,
+      kind,
+      sort
+    },
+    fetchPolicy: 'network-only'
+  });
+
+  const { i18n } = useTranslation();
+
+  interface ContentUiProps {
+    item: HydratedContentItem;
+    index?: number;
+  }
+
+  const DisplayListView = ({ item }: ContentUiProps) => {
+    return (
+      // list flex container
+      <LearnerAccessDisplayListView item={item} />
+    );
+  };
+
+  const DisplayGridView = ({ item }: ContentUiProps) => {
+    return (
+      // {/* course card */}
+      <LearnerAccessGridView item={item} />
+    );
+  };
+
+  if (loading) return <LoadingDots />;
+
+  if (error) return <>{error.message}</>;
+
+  if (!data || !data.UserContentItems) return <></>;
+
+  const listViewClassNames = gridViewActive
+    ? 'flex border rounded-l-md w-9 h-9 place-content-center items-center'
+    : 'flex border rounded-l-md w-9 h-9 place-content-center items-center bg-blue-600';
+
+  const gridViewClassNames = gridViewActive
+    ? 'flex border rounded-r-md w-9 h-9 place-content-center items-center bg-blue-600'
+    : 'flex border rounded-r-md w-9 h-9 place-content-center items-center';
+
+  return (
+    <>
+      {/* grid/list toggle */}
+      <div className="hidden sm:flex justify-end p-3">
+        {/* list display button */}
+        <button className={listViewClassNames} onClick={() => setGridActive(false)}>
+          <img src={listViewSelector} />
+        </button>
+
+        {/* grid display button */}
+        <button className={gridViewClassNames} onClick={() => setGridActive(true)}>
+          <img src={gridSelected} />
+        </button>
+      </div>
+      {gridViewActive ? (
+        <div className="grid gap-5 self-center grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
+          {data.UserContentItems.map(item => {
+            const hydratedItem = hydrateContent(i18n, item);
```

Comment:
Don't think we need to hydrate items both here and on L107. Think we can just iterate through the content items and hydrate them, assign 'em to an array, then iterate through the already hydrated array here and on L106

Reply:
possible solution pushed